hadoop-ozone/ozonefs-common/src/main/java/org/apache/hadoop/fs/ozone/BasicRootedOzoneFileSystem.java

```diff
@@ -723,6 +726,12 @@ public boolean mkdirs(Path f, FsPermission permission) throws IOException {
     return mkdir(f);
   }
 
+  @Override
+  public long getDefaultBlockSize() {
+    return (long) getConf().getStorageSize(
+        OZONE_SCM_BLOCK_SIZE, OZONE_SCM_BLOCK_SIZE_DEFAULT, StorageUnit.BYTES);
```

Comment:
Is this block size used by client while writing? In create and createNonRecursive it tends to ignore the specified block size. 
So, if it isn't correct, Isn't it same as not implementing it? `FileSystem` will anyway return some value, that can be tweaked as well by `fs.local.block.size`

Reply:
> In create and createNonRecursive it tends to ignore the specified block size.

Ozone also ignores it and uses uniform block size on server side (configurable via this `OZONE_SCM_BLOCK_SIZE` property).

> can be tweaked as well by `fs.local.block.size`

You are right, we could achieve the same result by forcing this config to the same value as SCM block size.  I think it's simpler and cleaner to override the method than to keep the config in sync.